docs/src/introduction/functions.md

```diff
@@ -0,0 +1,102 @@
+# Functions
+
+Though lists are a nice way of bundling code, they aren't perfect. For example, when creating your own named function, you will have to call it manually.
+
+```clojure
+'(1 +) 'add-one def
+
+1 add-one
+;; Pushes the list to the stack, but doesn't call it
+;; [1] -> [1 (1 +)]
+
+call
+;; Calls the list
+;; [1 (1 +)] -> [2]
+```
+
+## The `fn` Expression
+
+For this reason, Stack provides the `fn` expression to mark lists as functions. Internally, what you see as a "function" is actually just a list with the `fn` expression at the beginning.
+
+```clojure
+'(fn 1 +) 'add-one def
+
+1 add-one
+;; Pushes the list to the stack and calls it
+;; [1] -> [2]
+```
+
+Notice how we didn't need to call the list manually? That's because the `fn` expression tells Stack to call the list automatically. This is known as [auto-calling](/glossary.html#auto-calling).
+
+*Note: The evaluation pattern of functions is the same as lists: left to right, evaluating each expression and pushing it to the stack.*
+
+## Functions are Lists
+
+Because functions are just marked lists, you can still use them as lists. This means that you can build and modify functions at runtime.
+
+```clojure
+()
+'fn push
+1 push
+'+ push
+
```

Comment:
Include what the stack looks like at this point, so that the next comment can be derived.